src/main/java/com/carpool/car_pool/services/RideRequestService.java

```diff
@@ -126,4 +126,12 @@ public RideRequestResponse answerRideRequest(@Valid AnswerRideRequestRequest req
 
         return rideRequestConverter.entityToDTO(rideRequest);
     }
+
+    public List<RideRequestResponse> getRideRequestsForUser(UserEntity currentUser) {
+        List<RideRequestsEntity> rideRequests = rideRequestRepository.findByRequester(currentUser);
+
+        return rideRequests.stream()
```

Comment:
You could have rest of the function as one liner, similar to getRideRequestsForRideOffer return statement. Doesn't change much when it comes to memory or speed but it's just uniformity of coding that makes it easier to read